### ISSUE TITLE
Base image is way too old

### ISSUE THREAD

**Issue by FischerLGLN** [developmentseed/titiler#519]
#### Problem description
I like to use Titiler, but I noticed something:
The base images of tiangolo are 1 year old. Also, for Kubernetes, it is recommended to use normal Python images:
https://github.com/tiangolo/uvicorn-gunicorn-docker#-warning-you-probably-dont-need-this-docker-image

I also ran trivy image scans:

```sh
ghcr.io/developmentseed/titiler:latest (debian 11.0)
Total: 1719 (UNKNOWN: 6, LOW: 630, MEDIUM: 489, HIGH: 514, CRITICAL: 80)

titiler:python-3.9 (debian 11.5)
=========================
Total: 1178 (UNKNOWN: 1, LOW: 600, MEDIUM: 232, HIGH: 329, CRITICAL: 16)

titiler:python-3.10 (debian 11.5)
=========================
Total: 1178 (UNKNOWN: 1, LOW: 600, MEDIUM: 232, HIGH: 329, CRITICAL: 16)

titiler-bitnami:3.10 (debian 11.5)
Total: 347 (UNKNOWN: 1, LOW: 271, MEDIUM: 42, HIGH: 32, CRITICAL: 1)
```
#### Expected Output

You can reduce the number of vulnerabilities if you upgrade to an up-to-date base image. We actively and happily use the software, so I would like this to be fixed in a central place. If there is an alternative to issue creation for issues like this, I would look this up: https://github.com/developmentseed/titiler/security/policy   

Possible Dockerfile with bitnami/python:3.10:

Possible location and name: ```titiler/dockerfiles/Dockerfile.python-bitnami```
```Dockerfile
# Dockerfile for running titiler application with uvicorn server
ARG PYTHON_VERSION=3.10

FROM bitnami/python:${PYTHON_VERSION}
RUN apt update && apt upgrade -y \
  && rm -rf /var/lib/apt/lists/*

# Ensure root certificates are always updated at evey container build
# and curl is using the latest version of them
RUN mkdir /usr/local/share/ca-certificates/cacert.org
RUN cd /usr/local/share/ca-certificates/cacert.org && curl -k -O https://www.cacert.org/certs/root.crt
RUN cd /usr/local/share/ca-certificates/cacert.org && curl -k -O https://www.cacert.org/certs/class3.crt
RUN update-ca-certificates
ENV CURL_CA_BUNDLE /etc/ssl/certs/ca-certificates.crt

COPY src/titiler/ /tmp/titiler/
RUN pip install uvicorn /tmp/titiler/core /tmp/titiler/mosaic /tmp/titiler/application --no-cache-dir --upgrade
RUN rm -rf /tmp/titiler

ENV HOST 0.0.0.0
ENV PORT 80
CMD uvicorn titiler.application.main:app --host ${HOST} --port ${PORT}
```

The adjustment can then be made in the CI
https://github.com/developmentseed/titiler/blob/master/.github/workflows/ci.yml

Replace each occurrence of ```dockerfiles/Dockerfile.gunicorn``` with ```dockerfiles/Dockerfile.python-bitnami```.

#### Environment Information

Local Testing:

```sh
docker build -t titiler-bitnami:3.10 -f dockerfiles/Dockerfile.python-bitnami .
docker run -p 9015:8080 -it titiler-bitnami:3.10
```
I could open a pull request that makes the changes.

**Comment by vincentsarago** [comment]
what's the motivation for using bitnami over official python images ?

**Comment by FischerLGLN** [comment]
> what's the motivation for using bitnami over official python images ?
Most arguments are listed here: https://github.com/bitnami/containers#why-use-bitnami-images
Plus, it is not suffering from slow alpine based python code.

This is your built image for M1 development:
```sh
ghcr.io/vincentsarago/uvicorn-gunicorn:3.8 (debian 11.4)
========================================================
Total: 1221 (UNKNOWN: 2, LOW: 604, MEDIUM: 238, HIGH: 347, CRITICAL: 30)
```

In contrast to already posted:
```sh
titiler-bitnami:3.10 (debian 11.5)
Total: 347 (UNKNOWN: 1, LOW: 271, MEDIUM: 42, HIGH: 32, CRITICAL: 1)
```
My motivation is mainly secure usage of titiler in Kubernetes.
I could change the PR to have another image (the bitnami one) built with your pipeline, therefore not replacing your gunicorn image.
Then I can reference that image in my kubernetes yaml with every new version built.
But first its lunch time :wave:

**Comment by vincentsarago** [comment]
Thanks for the explanation 🙏 

I think it's fine moving both images to bitnami, and have both gunicorn and uvicorn published in titiler repo (and then update https://github.com/developmentseed/titiler/blob/master/deployment/k8s/charts/values.yaml#L4-L7 to point to the uvicorn one) 

Note: Can we just stay at python3.9 while we wait to have officially tested all the requirement in python3.10.

**Comment by vincentsarago** [comment]
closed by #520
